### PR TITLE
Adding admin dependency in authorization of dataset,files, metadata,search

### PR DESCRIPTION
**Pull request by ddey2**
For testing, please use endpoints on Postman or Clowder UI. Both should work.

### CODE REVIEW

**Comment by ddey2** [comment]
Hold on to review this one. I need to do some testing.

File: backend/app/tests/test_groups.py [hunk 1]
```diff
@@ -127,11 +127,6 @@ def test_member_permissions(client: TestClient, headers: dict):
     )
     assert response.status_code == 200
     assert response.json().get("id") is not None
-    response = client.get(
```
Comment (ddey2) [inline backend/app/tests/test_groups.py:130]: removed this test as there will always be a role for the admin user.

File: backend/app/tests/test_groups.py [hunk 2]
```diff
@@ -118,7 +118,7 @@ def test_member_permissions(client: TestClient, headers: dict):
     )
     assert response.status_code == 200
     assert response.json().get("id") is not None
-    assert response.json().get("role") == "viewer"
+    assert response.json().get("role") == "owner"
```
Comment (ddey2) [inline backend/app/tests/test_groups.py:121]: the admin user role will still be there for the dataset

**Review by lmarini — CHANGES_REQUESTED** [review]
The line `return DatasetDB.creator.email == current_username.email` at authentication.py:100 is returning `{'creator.email': '<current user email address>'}`  for some reason, which in turn is making all the `if admin` statements true, even if the user is not admin.

File: backend/app/routers/authorization.py [hunk 3]
```diff
@@ -68,18 +69,26 @@ async def save_authorization(
 async def get_dataset_role(
     dataset_id: str,
     current_user=Depends(get_current_username),
+    admin=Depends(get_admin),
 ):
     """Retrieve role of user for a specific dataset."""
+    # # admin is a superuser and has all the privileges
+    # if admin:
```
[the remark this answers is not in the document]
Comment (ddey2) [reply backend/app/routers/authorization.py:76]: I can't reproduce the issue. 
I created the 1st user and it was automatically assigned the admin. I created 2 datasets.
Screenshot 2023-11-13 at 2 43 55 PM


I created the 2nd user. I try to login. I can't see any dataset created by the admin,
Screenshot 2023-11-13 at 2 43 33 PM

File: backend/app/routers/authorization.py [hunk 4]
```diff
@@ -68,18 +69,26 @@ async def save_authorization(
...
+    #     return True
```
[the remark this answers is not in the document]
Comment (ddey2) [reply backend/app/routers/authorization.py]: done

File: backend/app/routers/authorization.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (ddey2) [reply backend/app/routers/authorization.py]: I reproduced the bug after an admin user shared the dataset with viewer role with another non-admin user.
Please check now.

**Comment by tcnichol** [comment]
Question on this one.

I have 2 users in my db right now. One is an admin, and one is not.

If I create a dataset with the non-admin user and upload files, the admin sees the dataset under explore, but when they click the dataset they do not see the files.

But if I download the dataset, they are downloaded.

If I try to access one of the files directly, this error is in the console:

`6565073147e9aeb9a3876b75:1 Access to fetch at 'http://localhost:8000/api/v2/authorizations/files/6565073147e9aeb9a3876b75/role' from origin 'http://localhost:3000' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource. If an opaque response serves your needs, set the request's mode to 'no-cors' to fetch the resource with CORS disabled.`

**Comment by ddey2** [comment]
@tcnichol Could you check the same issue here in this branch? https://github.com/clowder-framework/clowder2/pull/834
I see that you approved that PR. If we can merge the other one, it should be taken care of in this branch and we can merge it to main then. The super-admin branch contains a lot of other change too on top of this one which takes care of a lot of the functionalities.

**Review by longshuicy — CHANGES_REQUESTED** [review]
Test a few scenarios all works:
- Admin seeing other person's resources
- Admin search
- Non-Admin see resources only through sharing
- etc

If you fix the merge conflict we can merge :-)

**Comment by ddey2** [comment]
> Test a few scenarios all works:
> 
> * Admin seeing other person's resources
> * Admin search
> * Non-Admin see resources only through sharing
> * etc
> 
> If you fix the merge conflict we can merge :-)

done

**Review by longshuicy — APPROVED** [review]

**Pull request merged**